addons/mail/static/src/components/chat_window_manager/chat_window_manager_tests.js

```diff
@@ -554,57 +452,24 @@ QUnit.test('chat window: open / close', async function (assert) {
 });
 
 QUnit.test('chat window: close on ESCAPE', async function (assert) {
-    assert.expect(12);
+    assert.expect(10);
 
-    this.data['res.partner'].records = [{
-        email: "testpartnert@odoo.com",
-        id: 11,
-        name: "TestPartner",
-    }];
-    Object.assign(this.data.initMessaging, {
-        channel_slots: {
-            channel_channel: [{
-                channel_type: 'channel',
-                id: 20,
-                is_minimized: false,
-                is_pinned: true,
-                name: "General",
-                state: 'open',
-                uuid: 'channel-20-uuid',
-            }],
-        },
-    });
+    // expected partner to be found by mention during the test
+    this.data['res.partner'].records.push({ name: "TestPartner" });
+    // a chat window with thread is expected to be initially open for this test
+    this.data['mail.channel'].records.push({ is_minimized: true });
```

Comment:
Related to another comment, `push` in data really needs to be seen as `create` in python if we want to gravitate towards actually using the server some day.

And `id` should absolutely never be specified when doing `create`. I kept those that we needed (for bad reasons), but ideally we should have direct access to the `create` method in the test, so that we can receive the resulting `id` that was created if we need it in the test.

Reply:
As for `uuid`, same reasoning, it should never be set manually at `create`, there is a default value for that calling the appropriate crypto method.

Reply:
Makes sense when we use `create` in python, because it auto-generates an `id` and `uuid`, and you can get reference of record from `create`, so that you can assert matching `id`/`uuid` based on this record.

However, mock server seem to lack auto-generating `id`/`uuid`, and here you don't have a reference of server record from which you can assert it matches `id`/`uuid`... That's why you unfortunately have to hard-code `id` and `uuid` at the moment.

Reply:
Just to add another argument for explicit `id`/`uuid` at the moment: since data are incomplete, these tests may fail (e.g. flow in test calls another RPC, or mocked RPC received some changes), and it's painful to debug and know it comes from incompleteness of data, which require putting hard-coded `id`/`uuid` in the end.

Reply:
They are assigned a value. See `_getUnusedID` and `_applyDefaults`.

Reply:
Ok, so just to make sure I understand it: we should only set explicit `id` when we need to identify record in test based on id. For `uuid`, we could let default generate one and make use of hard-coded `id`. If we eventually use `create` in python, it returns the record itself (or its `id`), so that we never have to hard-code id again.

Reply:
Correct.

Note that this could be done in JS as well if we wanted, we just have to make the mock server (or a helper) accessible in the test, so that we could call a "createRecord" function and receive its result, instead of having to push data randomly without even knowing where it's going to be used without making sure data are passed to start, checking the start helper to see where the data is passed, etc.